### PR TITLE
[foreman] Obfuscate http_proxy passwords. PR-3878 improvement

### PR DESCRIPTION
**Pull request by pafernanr**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Comment by jcastill** [comment]
@pafernanr can you squash these two commits into one please?

File: sos/report/plugins/foreman.py [hunk 1]
```diff
@@ -328,16 +328,14 @@ def postproc(self):
             r"((\:|\s*)(passw|cred|token|secret|key).*(\:\s|=))(.*)",
             r'\1"********"')
         # hide proxy credentials..
-        self.do_paths_http_sub([
-            '/var/log/foreman/production.log*',
-        ])
```
[the remark this answers is not in the document]
Comment (pafernanr) [reply sos/report/plugins/foreman.py:333]: `do_paths_http_sub` function replaces only one occurrence of `r"(http(s)?://)\S+:\S+(@.*)", r"\1******:******\3")`.

In the foreman case there could be one or two occurrences on the same line. That's the reason to remove that function and improve the regex on foreman.py to capture one or multiple occurrences.

*_PROXY vars are being obfuscated at https://github.com/sosreport/sos/blob/main/sos/report/plugins/foreman.py#L299

[the remark this answers is not in the document]
Comment (pafernanr) [reply sos/report/plugins/foreman.py:333]: hmmm. I think the description for `do_paths_http_sub` is not accurate:
-  Obfuscate credentials in *_PROXY variables...

[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/report/plugins/foreman.py:333]: Yes, the description is a historic artifact from some first PR draft and should be updated.

[the remark this answers is not in the document]
Comment (jcastill) [reply sos/report/plugins/foreman.py:333]: @pmoravec can you help with this?

[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/report/plugins/foreman.py:333]: Indeed, the `do_paths_http_sub` can match just one instance on a given line - that is why my #3878 already obfuscates the logfile(s) twice. Which is ridiculous and this PR aims to improve it.

[the remark this answers is not in the document]
Comment (pafernanr) [reply sos/report/plugins/foreman.py:333]: If you agree, we might modify it to something like "Obfuscate BASIC_AUTH credentials in URLs for all files in the...". 
But again that function suffer the same issue. Only one match is being replaced /o\

[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/report/plugins/foreman.py:333]: I created https://github.com/sosreport/sos/pull/3882 for that. Usually, there is just one password on a line, so usually the method is sufficient, I think.

Or are there instances where we need to obfuscate a secret on the same line multiple times? This is generic question, not specific to the `do_paths_http_sub` method. As the method just calls something internal that is used everywhere, incl. by `do_path_regex_sub` method.

(i.e. should we utilize `repeatedly` option of `do_path_regex_sub`?)

**Review by pmoravec — APPROVED** [review]
Works well for me, tested against all instances of the original PR reproducer.

**Review by jcastill — APPROVED** [review]

**Review by arif-ali — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]
Cool! This makes my recent PR obsolete, I will drop it.

**Comment by arif-ali** [comment]
not for this PR, would we benefit on a test for multiple items on one line. I know I have the apt example in stagetwo already, which is a http_sub test for one item?

**Review by arif-ali — APPROVED** [review]

**Comment by pmoravec** [comment]
> not for this PR, would we benefit on a test for multiple items on one line.

Yeah we should add some test for either proxy credentials or for multiple passwords on a line. 

> I know I have the apt example in stagetwo already, which is a http_sub test for one item?

I dont understand you here.

**Comment by arif-ali** [comment]
> > I know I have the apt example in stagetwo already, which is a http_sub test for one item?
> 
> I dont understand you here.

i.e. there is a stagetwo test for apt, that checks for http obfuscation, but that just has single http credentials on one line :)

**Pull request merged**